### PR TITLE
Add timeout option

### PR DESCRIPTION
**Pull request by tomconnell-wf**
## Motivation
  
I hope to make the timeout on InfoReceiver handshake configurable.  https://github.com/sockjs/sockjs-client/pull/594

When that happens, the wrapper will need its api updated to actually pass in the timeout option to SockJS.

## Changes
  
Add the timeout to the SockJsOptions

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Architecture member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/sockjs_client_wrapper/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/sockjs_client_wrapper/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by tomconnell-wf** [comment]
This can't actually merge, since I manually made changes to sockjs.js, but the code changes are there because I needed to be able to test with https://github.com/Workiva/doc_plat_client/pull/30069

**Comment by aviary3-wk** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(1) Security relevant changes were detected</summary>
<li>Watched file <code>lib/sockjs.js</code> modified</li>
</details>

## Action Items
* Obtain a [security review](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); reviewer should pay special attention to insights listed above
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/client.dart [hunk 1]
```diff
@@ -187,10 +187,14 @@ class SockJSOptions {
   /// can be useful if you need to disable certain fallback transports.
   final List<String> transports;
 
+  /// The minimum timeout, in milliseconds.  If sockJS's internally calculated
+  /// timeout is higher, that will be used instead.
+  final int timeout;
```
Comment (patkujawa-wf) [inline lib/src/client.dart:192]: nit, thoughts?
```suggestion
  final int timeoutMs;
```

File: lib/sockjs.js [hunk 2]
```diff
@@ -547,19 +547,23 @@
...
     var self = this
       , url = urlUtils.addPath(baseUrl, '/info')
       ;
     debug('doXhr', url);
 
     this.xo = InfoReceiver._getReceiver(baseUrl, url, urlInfo);
 
+
+    if (timeout === undefined) {
+      timeout = InfoReceiver.timeout
```
Comment (patkujawa-wf) [inline lib/sockjs.js:560]: > If sockJS's internally calculated
>  /// timeout is higher, that will be used instead.
😕 if true, shouldn't this conditional be < ? But maybe also set a lower limit?

File: lib/src/client.dart (hunk 1 above)
Comment (tomconnell-wf) [reply lib/src/client.dart:192]: Sockjs calls it timeout, I decided to follow their api.  But, maybe clarity is better than congruency?

Comment (patkujawa-wf) [reply lib/src/client.dart:192]: Actually,
```suggestion
  final int minTimeoutMs;
```

File: lib/sockjs.js (hunk 2 above)
Comment (patkujawa-wf) [reply lib/sockjs.js:560]: Ah, it's *min* timeout!

File: lib/src/client.dart (hunk 1 above)
Comment (patkujawa-wf) [reply lib/src/client.dart:192]: Meh, if they use timeout, let's use timeout.

File: lib/sockjs.js (hunk 2 above)
Comment (tomconnell-wf) [reply lib/sockjs.js:560]: Oh, whoops, I must have had that slip through the cracks somewhere.  I'll see what the sockjs maintainer says.

**Review by patkujawa-wf — APPROVED** [review]

**Comment by patkujawa-wf** [comment]
Looks like `dart format` needed too

**Comment by tomconnell-wf** [comment]
Couldn't get sockjs to look at my PR.  https://github.com/sockjs/sockjs-client/pull/594

**Pull request closed without merging**